core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/ApplyPreferredTableExecutePartitioning.java

```diff
@@ -58,10 +62,14 @@ public Result apply(TableExecuteNode node, Captures captures, Context context)
             return enable(node);
         }
 
-        double expectedNumberOfPartitions = getRowsCount(
-                context.getStatsProvider().getStats(node.getSource()),
-                node.getPreferredPartitioningScheme().get().getPartitioning().getColumns());
+        PlanNodeStatsEstimate stats = context.getStatsProvider().getStats(node.getSource());
+        Set<Symbol> partitionColumns = node.getPreferredPartitioningScheme().get().getPartitioning().getColumns();
+        if (!canEstimateNumberOfPartitions(stats, partitionColumns)) {
+            // Force 'preferred write partitioning' when stats are missing
+            return enable(node);
```

Comment:
This is some kind of counterintuitive with `getRowsCount` result and `isNaN` where we return `Result.empty` instead of `enable(node)`. Also, we ignore `minimumNumberOfPartitions`.

Reply:
We don't have to check isNan on expectedNumberOfPartition since we are already checking that inside `canEstimateNumberOfPartitions`. So, I've removed that

Reply:
> Also, we ignore minimumNumberOfPartitions.

Yes, the core of this pr is that we want to ignore `minimumNumberOfPartitions` if there are no stats. That's because if there are too many partitions and we don't go through the preferred partition route due to lack of statistics then it could result in OOM error thus not reliable.